src/main/java/fi/metatavu/metamind/server/rest/StoriesApiImpl.java

```diff
@@ -281,18 +286,24 @@ public Response createSession(fi.metatavu.metamind.rest.model.Session body, UUID
   public Response createStory(Story body) {
     UUID loggedUserId = getLoggerUserId();
     Locale locale = LocaleUtils.toLocale(body.getLocale());
+    List<AuthorizationScope> authorisationScopesList = Arrays.asList(AuthorizationScope.STORY_ACCESS, AuthorizationScope.STORY_MANAGE);
+    fi.metatavu.metamind.persistence.models.Story story = storyController.createStory(locale, body.getName(), body.getDafaultHint(), loggedUserId);
+    try {
+      UUID createdResourceId = authenticationController.createProtectedResource(loggedUserId, String.format(STORY_NAME_TEMPLATE, story.getId()), String.format("/v2/stories/%s", story.getId()), "story", authorisationScopesList);
+      List<UUID> policyIds = authenticationController.checkUserPolicy(loggedUserId);
+      authenticationController.upsertScopePermission(createdResourceId, authorisationScopesList, String.format("Permission for story-%s", story.getId()), DecisionStrategy.AFFIRMATIVE, policyIds);
+    } catch (AuthorizationException e) {
+      
+      return createInternalServerError(String.format(EXCEPTION_CAUGHT, e.getMessage()));
```

Comment:
It would be better to use more descriptive error message, e.g. "Failed to apply authorization rules for story: %s"